Dockerfile

```diff
@@ -0,0 +1,93 @@
+                   which                   \
+                   &&                      \
+    yum clean all
+
+# Add persistent data volume for postgres
+VOLUME [ "/var/opt/rh/rh-postgresql94/lib/pgsql/data" ]
+
+# Download chruby and chruby-install, install, setup environment, clean all
+RUN curl -sL https://github.com/postmodern/chruby/archive/v0.3.9.tar.gz | tar xz && \
+    cd chruby-0.3.9 && make install && scripts/setup.sh && \
+    echo "gem: --no-ri --no-rdoc --no-document" > ~/.gemrc && \
+    echo "source /usr/local/share/chruby/chruby.sh" >> ~/.bashrc && \ 
+    curl -sL https://github.com/postmodern/ruby-install/archive/v0.6.0.tar.gz | tar xz && \
+    cd ruby-install-0.6.0 && make install && ruby-install ruby 2.2.4 -- --disable-install-doc && \
+    echo "chruby ruby-2.2.4" >> ~/.bash_profile && \
+    rm -rf ~/ruby* && rm -rf /usr/local/src/* && yum clean all
+
+## Environment for scripts
+RUN echo "export BASEDIR=/manageiq" > /etc/default/evm && \
+    echo "export PATH=$PATH:/opt/rubies/ruby-2.2.4/bin:/opt/rh/rh-postgresql94/root/bin:/opt/rh/rh-postgresql94/root/usr/libexec" >> /etc/default/evm && \
+    echo "[[ -s /etc/default/evm_postgres ]] && source /etc/default/evm_postgres" >> /etc/default/evm && \
+    echo "export APPLIANCE_PG_SCL_NAME=rh-postgresql94" > /etc/default/evm_postgres && \
+    echo "export APPLIANCE_PG_SERVICE=rh-postgresql94-postgresql" >> /etc/default/evm_postgres && \
+    echo "export APPLIANCE_PG_DATA=/var/opt/rh/rh-postgresql94/lib/pgsql/data" >> /etc/default/evm_postgres && \
+    echo "export APPLIANCE_PG_CHECKDB=/opt/rh/rh-postgresql94/root/usr/libexec/postgresql-check-db-dir" >> /etc/default/evm_postgres && \
+    echo "[[ -s /opt/rh/\${APPLIANCE_PG_SCL_NAME}/enable ]] && source /opt/rh/\${APPLIANCE_PG_SCL_NAME}/enable" >> /etc/default/evm_postgres
+
+## Create basedir, GIT clone miq (shallow)
+#RUN mkdir -p /manageiq && git clone --depth 1 https://github.com/ManageIQ/manageiq /manageiq
+RUN mkdir -p /manageiq
+ADD . /manageiq
+
+## Change WORKDIR to clone dir, copy docker_setup, start all, docker_setup, shutdown all, clean all
+WORKDIR /manageiq
+COPY docker-assets/docker_setup bin/docker_setup
+RUN /bin/bash -l -c "/usr/bin/memcached -u memcached -p 11211 -m 64 -c 1024 -l 127.0.0.1 -d && \
+    source /etc/default/evm && \
+    bin/docker_setup --no-db --no-tests && \
+    pkill memcached && \
+    rm -rvf /opt/rubies/ruby-2.2.4/lib/ruby/gems/2.2.0/cache/*"
+
+## Copy db init script, evmserver startup script and systemd evmserverd unit file
+COPY docker-assets/docker_initdb bin/docker_initdb
+COPY docker-assets/evmserver.sh bin/evmserver.sh
+COPY docker-assets/evmserverd.service /usr/lib/systemd/system/evmserverd.service
+
+## Scripts symblinks
+RUN ln -s /manageiq/bin/evmserver.sh /usr/bin && \
+    ln -s /manageiq/bin/docker_initdb /usr/bin
+
+## Enable services on systemd
+RUN systemctl enable evmserverd memcached
+
+## Expose required container ports
+EXPOSE 3000 4000
```

Comment:
@bdunne  I answered the above comment in that discussion.

IIUC the only relevant ports still not opened are the HTML5 console, and it could still be opened by the docker run command.

Please reply back in case one of my answers are not clear.